set.mm

```diff
@@ -147154,6 +147148,405 @@ of an open range of nonnegative integers (of length equal to the length of
+    VKVMAVMAAGBVCVDVBVEVFVG $.
+
+  $( Length of a prefix.  (Contributed by Stefan O'Rear, 24-Aug-2015.)
+     (Revised by AV, 2-May-2020.) $)
+  pfxlen $p |- ( ( S e. Word A /\ L e. ( 0 ... ( # ` S ) ) )
+                    -> ( # ` ( S prefix L ) ) = L ) $=
+    ( cword wcel cc0 chash cfv cfz co wa cpfx cfzo wfn pfxfn hashfn syl elfznn0
+    wceq cn0 adantl hashfzo0 eqtrd ) BADEZCFBGHZIJEZKZBCLJZGHZFCMJZGHZCUGUHUJNU
+    IUKSBCAOUJUHPQUGCTEZUKCSUFULUDCUERUACUBQUC $.
+
+  ${
+    $d x A $.  $d x S $.
+    $( A word is a prefix of itself.  (Contributed by Stefan O'Rear,
+       16-Aug-2015.)  (Revised by AV, 2-May-2020.) $)
+    pfxid $p |- ( S e. Word A -> ( S prefix ( # ` S ) ) = S ) $=
+      ( vx cword wcel cc0 chash cfv cfzo co cpfx cfz wf lencl nn0fz0 sylib pfxf
+      cn0 mpdan ffnd wrdfn cv wa wceq simpl adantr simpr pfxfv syl3anc eqfnfvd
+      ) BADEZCFBGHZIJZBULKJZBUKUMAUNUKULFULLJEZUMAUNMUKULREUOABNULOPZULABQSTABU
+      AUKCUBZUMEZUCUKUOURUQUNHUQBHUDUKURUEUKUOURUPUFUKURUGUQULABUHUIUJ $.
+  $}
+
+  $( The range of a prefix of a word is a subset of the set of symbols for the
+     word.  (Contributed by AV, 2-May-2020.) $)
+  pfxrn $p |- ( ( W e. Word V /\ L e. ( 0 ... ( # ` W ) ) )
+                 -> ran ( W prefix L ) C_ V ) $=
+    ( cword wcel cc0 chash cfv cfz co wa cfzo cpfx pfxf frnd ) CBDEAFCGHIJEKFAL
+    JBCAMJABCNO $.
+
+  $( A prefix consisting of at least one symbol is not empty.  (Contributed by
+     Alexander van der Vekens, 4-Aug-2018.)  (Revised by AV, 2-May-2020.) $)
+  pfxn0 $p |- ( ( W e. Word V /\ L e. NN /\ L <_ ( # ` W ) )
+                 -> ( W prefix L ) =/= (/) ) $=
+    ( cword wcel cn chash cfv cle wbr w3a cpfx co c0 wne cc0 cfzo 3ad2ant2 wceq
+    cn0 lbfzo0 ne0i sylbir wf wb cfz simp1 nnnn0 lencl 3ad2ant1 simp3 syl3anbrc
+    elfz2nn0 pfxf syl2anc f0dom0 bicomd syl necon3bid mpbird ) CBDEZAFEZACGHZIJ
+    ZKZCALMZNOPAQMZNOZVBVAVHVDVBPVGEVHAUAVGPUBUCRVEVFNVGNVEVGBVFUDZVFNSZVGNSZUE
+    VEVAAPVCUFMEZVIVAVBVDUGVEATEZVCTEZVDVLVBVAVMVDAUHRVAVBVNVDBCUIUJVAVBVDUKAVC
+    UMULABCUNUOVIVKVJVFVGBUPUQURUSUT $.
+
+  $( The value of the prefix extractor is the empty set (undefined) if the
+     argument is not within the range of the word.  (Contributed by AV,
+     3-May-2020.) $)
+  pfxnd $p |- ( ( W e. Word V /\ L e. NN0 /\ ( # ` W ) < L )
+                 -> ( W prefix L ) = (/) ) $=
+    ( cword wcel cn0 chash cfv clt wbr w3a cpfx co cc0 cop csubstr c0 pfxval cz
+    wceq 3adant3 cle w3o simp1 0zd nn0z 3ad2ant2 3jca 3mix3 3ad2ant3 sylc eqtrd
+    swrdnd ) CBDZEZAFEZCGHAIJZKZCALMZCNAOPMZQUOUPUSUTTUQCAUNRUAURUONSEZASEZKNNI
+    JZANUBJZUQUCZUTQTURUOVAVBUOUPUQUDURUEUPUOVBUQAUFUGUHUQUOVEUPUQVCVDUIUJNABCU
+    MUKUL $.
+
+  $( Length of a prefix of a word reduced by a single symbol.  (Contributed by
+     AV, 4-Aug-2018.)  (Revised by AV, 3-May-2020.) $)
+  pfxlen0 $p |- ( ( W e. Word V /\ L e. NN0 /\ ( # ` W ) = ( L + 1 ) )
+                  -> ( # ` ( W prefix L ) ) = L ) $=
```

Comment:
isn't it enough to suppose `L <= length(W)` ?

Reply:
> isn't it enough to suppose `L <= length(W)` ?

This is covered by ~ pfxlen.

But you are right, ~pfxlen0 makes no sense, and its predecessor swrd0len0(OLD) is only used once. Therefor I will delete ~pfxlen0.

Reply:
Done